### ISSUE TITLE
Improve `AssetEvent` docs

### ISSUE THREAD

**Issue by alercah** [bevyengine/bevy#11943]
The docs for [`AssetEvent`](https://docs.rs/bevy/latest/bevy/asset/prelude/enum.AssetEvent.html) are a bit vague on what exactly it means for an asset to be "added", or "modified" or "loaded with dependencies".

For instance, when hot-reloading an asset, we expect `Modified` to fire, but what about `LoadedWithDependencies`? Will that fire? What if the newly modified asset has additional dependencies?

It should be easy to deduce exactly when an asset event will fire for a particular sequence of asset operations.

**Comment by alercah** [comment]
Another example: Do `Unused` assets not get automatically `Removed`? When would one want to use one over the other?

**Comment by JMS55** [comment]
Unused assets will get removed automatically, _if they were not already removed earlier_. Unused is mainly for RenderAsset and RenderAssetUsage, so that we can remove the asset data from the Assets collection in RAM, but know when to unload the corresponding GPU data in VRAM when the asset is no longer actually used.

Tldr Unused = last handle dropped, Removed = removed from Assets<T> resource.
